src/System.Resources.Reader/src/System/Resources/ResourceReader.cs

```diff
@@ -19,17 +18,25 @@ public sealed class ResourceReader : IDisposable
         private long _dataSectionOffset;  // Offset to Data section of file.
 
         private int[] _namePositions; // relative locations of names
-        private Type[] _typeTable;    // Lazy array of Types for resource values.
-        private int[] _typeNamePositions;  // To delay initialize type table
+        private int _stringTypeIndex;
+        private int _numOfTypes;
         private int _numResources;    // Num of resources files, in case arrays aren't allocated.      
         private int _version;
 
+        // "System.String, mscorlib" representation in resources stream
+        private static byte[] s_SystemStringName = new byte[] { 23,
+            (byte)'S', (byte)'y', (byte)'s', (byte)'t', (byte)'e', (byte)'m', (byte)'.',
+            (byte)'S', (byte)'t', (byte)'r', (byte)'i', (byte)'n', (byte)'g', (byte)',', (byte)' ',
+            (byte)'m', (byte)'s', (byte)'c', (byte)'o', (byte)'r', (byte)'l', (byte)'i', (byte)'b' };
```

Comment:
What if string isn't defined in mscorlib?  e.g. if/when this library is used with https://github.com/dotnet/corert/blob/master/src/Runtime.Base/src/System/String.cs ?